src/react.force.smartstore.ts

```diff
@@ -0,0 +1,757 @@
+    storeName: storeConfig.storeName,
+  };
+  execArgs[
+    entryIdsOrQuerySpec instanceof Array ? "entryIds" : "querySpec"
+  ] = entryIdsOrQuerySpec;
+  execArgs[
+    entryIdsOrQuerySpec instanceof Array ? "querySpec" : "entryIds"
+  ] = null;
+  exec(successCB, errorCB, "removeFromSoup", execArgs);
+};
+
+// ====== Cursor manipulation ======
+export const moveCursorToPageIndex = <T>(
+  storeConfig: StoreConfig | boolean,
+  cursor: StoreCursor<T>,
+  newPageIndex: number,
+  successCB: (result: T) => void,
+  errorCB: (err: Error) => void
+): void => {
+  storeConfig = checkFirstArg(storeConfig);
+
+  // query returns serialized json on iOS starting in 7.0
+  let successCBdeserializing;
+
+  if (successCB) {
+    successCBdeserializing = (result: any) =>
+      successCB(typeof result === "string" ? safeJSONparse(result) : result);
+  } else {
+    successCBdeserializing = successCB;
+  }
+
+  exec(successCBdeserializing, errorCB, "moveCursorToPageIndex", {
+    cursorId: cursor.cursorId,
+    index: newPageIndex,
+    isGlobalStore: storeConfig.isGlobalStore,
+    storeName: storeConfig.storeName,
+  });
+};
+
+export const moveCursorToNextPage = <T>(
+  storeConfig: StoreConfig | boolean,
+  cursor: StoreCursor<T>,
+  successCB: (result: T) => void,
+  errorCB: (cursor: StoreCursor<T>, err: Error) => void
+): void => {
+  storeConfig = checkFirstArg(storeConfig);
+
+  const newPageIndex = cursor.currentPageIndex + 1;
+  if (newPageIndex >= cursor.totalPages) {
+    errorCB(
+      cursor,
+      new Error("moveCursorToNextPage called while on last page")
+    );
+  } else {
+    moveCursorToPageIndex(
+      storeConfig,
+      cursor,
+      newPageIndex,
+      successCB,
+      errorCB
+    );
+  }
+};
+
+export const moveCursorToPreviousPage = <T>(
+  storeConfig: StoreConfig | boolean,
+  cursor: StoreCursor<T>,
+  successCB: (result: T) => void,
+  errorCB: (cursor: StoreCursor<T>, err: Error) => void
+): void => {
+  storeConfig = checkFirstArg(storeConfig);
+
+  const newPageIndex = cursor.currentPageIndex - 1;
+  if (newPageIndex < 0) {
+    errorCB(
+      cursor,
+      new Error("moveCursorToPreviousPage called while on first page")
+    );
+  } else {
+    moveCursorToPageIndex(
+      storeConfig,
+      cursor,
+      newPageIndex,
+      successCB,
+      errorCB
+    );
+  }
+};
+
+export const closeCursor = <T>(
+  storeConfig: StoreConfig | boolean,
+  cursor: StoreCursor<T>,
+  successCB: (result: T) => void,
+  errorCB: (err: Error) => void
+): void => {
+  storeConfig = checkFirstArg(storeConfig);
+
+  exec(successCB, errorCB, "closeCursor", {
+    cursorId: cursor.cursorId,
+    isGlobalStore: storeConfig.isGlobalStore,
+    storeName: storeConfig.storeName,
+  });
+};
+
+// ====== Store Operations ======
+export const getAllStores = <T>(
+  successCB: (result: T) => void,
+  errorCB: (err: Error) => void
+): void => {
+  exec(successCB, errorCB, "getAllStores", {});
+};
+
+export const getAllGlobalStores = <T>(
```

Comment:
Returns an array of dictionaries. [ {storeName:'' , isGlobalStore: true or false},{}]